shell/common/rasterizer.cc

```diff
@@ -127,9 +127,12 @@ void Rasterizer::Draw(fml::RefPtr<Pipeline<flutter::LayerTree>> pipeline) {
   // if the raster status is to resubmit the frame, we push the frame to the
   // front of the queue and also change the consume status to more available.
   if (raster_status == RasterStatus::kResubmit) {
-    auto front_continuation = pipeline->ProduceToFront();
-    front_continuation.Complete(std::move(resubmitted_layer_tree_));
-    consume_result = PipelineConsumeResult::MoreAvailable;
+    auto front_continuation = pipeline->ProduceIfEmpty();
+    bool result =
+        front_continuation.Complete(std::move(resubmitted_layer_tree_));
+    if (result) {
+      consume_result = PipelineConsumeResult::MoreAvailable;
+    }
```

Comment:
It seems to me that if `Complete` returns false here, then there must be something in the queue, so the call to `Complete` just above this (line 126?) would probably have already returned `MoreAvailable` anyway?  Am I missing something?

Reply:
Oh, you are right!  Reverting.